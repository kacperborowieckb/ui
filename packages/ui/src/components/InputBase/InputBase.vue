<template>
  <input
    v-model="model"
    class="placeholder:text-muted-foreground py-1/2 h-full min-w-[120px] flex-1 px-2 focus:outline-0"
    type="text"
    :placeholder="placeholder"
    :disabled="disabled"
  >
</template>

<script lang="ts">
export interface InputBaseProps {
  placeholder?: string
  disabled?: boolean
}
</script>

<script setup lang="ts">
defineProps<InputBaseProps>()

const model = defineModel<string>()
</script>
