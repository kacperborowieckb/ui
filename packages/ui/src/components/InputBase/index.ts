export {
  default as InputBase,
  type InputBaseProps,
} from './InputBase.vue'
