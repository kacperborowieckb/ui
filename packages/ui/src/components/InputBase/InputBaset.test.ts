import { shallowMount } from '@vue/test-utils'

import InputBase from './InputBase.vue'

describe('inputBase', () => {
  it('should render correctly', () => {
    const wrapper = shallowMount(InputBase, {
      props: {
        placeholder: 'mockPlaceholder',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should reflect disabled state', () => {
    const wrapper = shallowMount(InputBase, {
      props: {
        disabled: true,
      },
    })

    expect(wrapper.find('input').attributes('disabled')).toBeDefined()
  })

  it('should correctly define model', () => {
    const wrapper = shallowMount(InputBase)

    wrapper.find('input').setValue('mockValue')

    expect(wrapper.emitted('update:modelValue')).toEqual([['mockValue']])
  })
})
