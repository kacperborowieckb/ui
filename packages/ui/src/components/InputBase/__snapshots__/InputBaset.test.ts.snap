// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inputBase > should render correctly 1`] = `"<input class="placeholder:text-muted-foreground py-1/2 h-full min-w-[120px] flex-1 px-2 focus:outline-0" type="text" placeholder="mockPlaceholder">"`;
